### PR TITLE
Correct C-c and C-h behaviour 

### PR DESCRIPTION
**Pull request by 0atman**
Corrected C-c and C-h behaviour  using `controlCHook()` and `readLine` exceptions

### CODE REVIEW

**Comment by 0atman** [comment]
What do you think of this? @Tangdongle

File: src/inim.nim [hunk 1]
```diff
@@ -197,6 +194,19 @@ proc showError(output: string) =
...
+    result = "nim> "
+    previouslyIndented = false
+  else:
+    result = ".... "
+    # Auto-indent (multi-level)
+    result &= indentSpaces.repeat(indentLevel)
+
+proc controlCHook() {.noconv.} =
+  stdout.write("\n" & getPromptSymbol())
+  stdout.flushFile()
```
Comment (0atman) [inline src/inim.nim:208]: This guy is new

File: src/inim.nim [hunk 2]
```diff
@@ -239,14 +241,14 @@ proc hasIndentTrigger*(line: string): bool =
...
         try:
-            currentExpression = readLineFromStdin(getPromptSymbol()).strip
-        except IOError:
-            bufferRestoreValidCode()
-            indentLevel = 0
-            tempIndentCode = ""
-            continue
+          stdout.write(getPromptSymbol())
+          stdout.flushFile()
+          currentExpression = stdin.readLine()
```
Comment (0atman) [inline src/inim.nim:249]: Can you think why `readLineFromStdin` was used before? My solution seems almost TOO simple...

Comment (Tangdongle) [reply src/inim.nim:249]: I think the main reason is to leverage the features from `linenoise`, like navigating with the arrow keys

Comment (0atman) [reply src/inim.nim:249]: oh! Well spotted. Can we wrap this simply with `rlwrap`? That does the trick, but we'd need to vendor that in...

**Comment by 0atman** [comment]
this is the wrong way solve this problem

**Pull request closed without merging**